### PR TITLE
Record: SP8192 + SLOT-4 + TTT + 3-Layer Recurrence + Parallel Residuals (1.0616 BPB)

### PR DESCRIPTION
**Pull request by powerpratik**
## Summary

  - **val_bpb: 1.0616** (3-seed mean, std 0.0005) on 8xH100 80GB HBM3
  - Beats merged SOTA (PR #1493, 1.0810 BPB) by **-0.0194 BPB**
  - Training completes in <600s, artifact ~16.0MB

  ## Key Contribution: SLOT (Sample-Level Optimization at Test-time)

  Adds per-window logit bias optimization at eval time on top of the PR #1493 stack. For each
  sliding window, a zero-initialized delta tensor is optimized via 4 AdamW steps (lr=0.01,
  wd=0.01) to minimize cross-entropy, then discarded. This captures local distributional patterns
   the model misses.

  ## 3-Seed Results

  | Seed | Sliding BPB | TTT BPB | **SLOT BPB** |
  |------|-------------|---------|-------------|
  | 314 | 1.0839 | 1.0827 | **1.0611** |
  | 42 | 1.0838 | 1.0832 | **1.0617** |
  | 999 | 1.0838 | 1.0832 | **1.0622** |
  | **Mean** | 1.0838 | 1.0830 | **1.0616** |

  ## Reproduction

  ```bash
  SEED=314 SLOT_ENABLED=1 SLOT_STEPS=4 SLOT_LR=0.01 TTT_ENABLED=1 \
    torchrun --standalone --nproc_per_node=8 train_gpt.py


  Attribution

  Built on PR #1493 (@bigbag) stack. SLOT implementation is novel to this submission.
  - SP8192 + GPTQ SDClip: @clarkkev (PR #1394)
  - Depth recurrence: @dexhunter (PR #1331, #1437)
  - Parallel residuals: @Robby955 (PR #1412), @msisovic (PR #1204)
  - Legal TTT: @abaybektursun (PR #549), @dexhunter (PR #1413)

### CODE REVIEW

**Comment by regina-openai** [comment]
Thanks for submitting! However, because the artifact size is 16001154 bytes (which is over the requirement of 16000000 bytes), this submission is ineligible for the record leaderboard.

**Comment by powerpratik** [comment]
Hello Regina,

Thank you for the update. I completely understand the stated requirement, but I wanted to reach out because my submission exceeded the limit by only 1,154 bytes — an incredibly small margin relative to the overall file size.
Given how close it was, I was wondering if there’s any possibility of an exception, manual review, or resubmission opportunity. I put a huge amount of effort into the artifact, and it would mean a lot to still be considered for the leaderboard if there’s any flexibility at all.
I fully respect the rules either way, but I figured it was worth asking considering how narrowly I missed the cutoff.
Thank you again for your time and consideration.

Bests,
Pratik Poudel


From: regina-openai ***@***.***>
Date: Monday, 27 April 2026 at 19:23
To: openai/parameter-golf ***@***.***>
Cc: Pratik Poudel ***@***.***>; Author ***@***.***>
Subject: Re: [openai/parameter-golf] Record: SP8192 + SLOT-4 + TTT + 3-Layer Recurrence + Parallel Residuals (1.0616 BPB) (PR #1647)

[https://avatars.githubusercontent.com/u/188402210?s=20&v=4]regina-openai left a comment (openai/parameter-golf#1647)<https://github.com/openai/parameter-golf/pull/1647#issuecomment-4331181039>

Thanks for submitting! However, because the artifact size is 16001154 bytes (which is over the requirement of 16000000 bytes), this submission is ineligible for the record leaderboard.
